pkg/kapp/resources/identified_resources_list.go

```diff
@@ -36,6 +37,10 @@ func (r IdentifiedResources) List(labelSelector labels.Selector, resRefs []Resou
 		schema.GroupVersionResource{Version: "v1", Resource: "componentstatuses"},
 	})
 
+	if opts.GKsScope != nil {
```

Comment:
I do believe I have used `nil` and empty slices inter changeably. We could definitely switch it up to use empty slices everywhere instead,
Maybe we could check on length instead.

Reply:
The ResourceRef struct is heavily based on GVR matching where as our focus is on GKs. Since we cannot obtain GVRs from manifests reliably, it does not let us scope aggressively.

Reply:
Will verify that we are comparing slices appropriately everywhere

Reply:
Went through the places where we were comparing this slice and switched to comparing lengths instead 🚀